interfaces/builtin/ssh-control.go

```diff
@@ -0,0 +1,39 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const sshControlConnectedPlugAppArmor = `
+# Description: Allow modifying the system SSH configuration
+
+/etc/ssh/sshd_not_to_be_run rw,
+`
+
+// NewSshControlInterface returns a new "ssh-control" interface.
+func NewSshControlInterface() interfaces.Interface {
+	return &commonInterface{
+		name: "ssh-control",
```

Comment:
https://bugs.launchpad.net/snappy/+bug/1606574 also discusses ssh from the perspective of the user and suggests 'ssh-control' as the interface name. As such, I think we need to think carefully through the name. I'll put up a strawman proposal to kick off discussion:

- default template - gives read/write access to /home/<user>/snap/$SNAP_NAME/$SNAP_REVISION/.ssh: **DONE**
- ssh-observe - give read access to /home/<user>/.ssh/*
- ssh-control - give read/write access to /home/<user>/.ssh/*
- sshd-control - give read/write access to /etc/ssh/ssh{,d}_config and /etc/ssh/sshd_not_to_be_run

Now, like with this PR, the actual accesses may change some but the above descriptions and intent of the interface should be clear as we move forward.

@morphis and @niemeyer - what do you think?